layer/ro_layer.go

```diff
@@ -43,6 +43,22 @@ func (rl *roLayer) TarStream() (io.ReadCloser, error) {
 	return rc, nil
 }
 
+func (rl *roLayer) TarStreamFrom(parent ChainID) (io.ReadCloser, error) {
```

Comment:
`TarStream` on `roLayer` must guarantee that the returned tar stream matches exactly what was given on register. Since this tar stream was never registered we cannot make that guarantee nor is it important. We should document these guarantees for `roLayer` so it is not attempted to be used for a push, but this looks correct to me.